Vostok.Metrics.System/Helpers/TeamingCollector.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+using Newtonsoft.Json.Linq;
+
+namespace Vostok.Metrics.System.Helpers
+{
+    // Specification: https://github.com/jpirko/libteam/wiki/Infrastructure-Specification
+    // Source code: https://github.com/jpirko/libteam/blob/master/libteamdctl/libteamdctl.c
+    internal class TeamingCollector : IDisposable
+    {
+        private readonly IntPtr teamdctlPointer;
+
+        public TeamingCollector(IntPtr ctl, string teamingInterfaceName)
+        {
+            teamdctlPointer = ctl;
+
+            if (teamdctl_connect(ctl, teamingInterfaceName, null, null) != 0)
+                throw new Exception("Unable to connect to teaming driver. Note that this operation is possible only under sudo.");
+        }
+
+        public IEnumerable<string> GetChildPorts()
+        {
+            var config = Marshal.PtrToStringAuto(teamdctl_state_get_raw(teamdctlPointer));
```

Comment:
I've noticed that `teamdctl_state_get_raw` returns a pointer to unmanaged memory. Who's going to free it? The doc for `Marshal.PtrToStringAuto` states that it's not responsible for that.

Reply:
I did it on purpose.
It is going to be freed by the teamdctl library.

Reply:
Ha, interesting. I wonder how the library knows when you've finished using the returned pointer. What if you decide to access it 2 hours from this call? Or does it simply reuse the same pooled memory block on each call internally?

Reply:
I'm not sure currently.
I assume that the memory is going to be freed when you call "teamdctl_disconnect", but let's find out!

Reply:
Ok, it was pretty simple.

Inside the method some kind of `cache` is used. 
Returned string comes from this cache. 

It is indeed reuse the same pooled memory block on each call. (Each call of the same method, actually).

And the memory is freed when cache is cleaned manually or the whole driver connector is freed. (`teamdctl_free`).